providers/implementations/macs/cmac_prov.c

```diff
@@ -101,8 +101,12 @@ static void *cmac_dup(void *vsrc)
 static size_t cmac_size(void *vmacctx)
 {
     struct cmac_data_st *macctx = vmacctx;
+    const EVP_CIPHER_CTX *cipherctx = CMAC_CTX_get0_cipher_ctx(macctx->ctx);
 
-    return EVP_CIPHER_CTX_get_block_size(CMAC_CTX_get0_cipher_ctx(macctx->ctx));
+    if (EVP_CIPHER_CTX_get0_cipher(cipherctx) == NULL)
+        return 0;
+
+    return EVP_CIPHER_CTX_get_block_size(cipherctx);
```

Comment:
Is this needed, given that EVP_CIPHER_CTX_get_block_size just calls EVP_CIPHER_get_block_size, which you've modified above to return 0 if cipher is NULL?

Reply:
Yes, it is needed since my last commit reverts the initial change I made to `EVP_CIPHER_get_block_size()` (as per the suggestion from Tomas and Matt).